Modules/Loadable/Segmentations/EditorEffects/qSlicerSegmentEditorPaintEffect.cxx

```diff
@@ -1558,49 +1536,38 @@ void qSlicerSegmentEditorPaintEffect::updateGUIFromMRML()
   bool brushDiameterIsRelative = this->integerParameter("BrushDiameterIsRelative");
 
   d->BrushDiameterRelativeToggle->blockSignals(true);
-  d->BrushDiameterRelativeToggle->setText(brushDiameterIsRelative ? "%" : tr("mm"));
+  d->BrushDiameterRelativeToggle->setChecked(!brushDiameterIsRelative);
   d->BrushDiameterRelativeToggle->blockSignals(false);
 
-  d->BrushDiameterSlider->blockSignals(true);
+  d->BrushDiameterSliderWidget->blockSignals(true);
   if (brushDiameterIsRelative)
     {
-    d->BrushDiameterSlider->setMinimum(1);
-    d->BrushDiameterSlider->setMaximum(25);
-    d->BrushDiameterSlider->setValue(this->doubleParameter("BrushRelativeDiameter"));
-    d->BrushDiameterSlider->setSingleStep(1);
+    d->BrushDiameterSliderWidget->setRange(1, 25);
+    d->BrushDiameterSliderWidget->setValue(this->doubleParameter("BrushRelativeDiameter"));
+    d->BrushDiameterSliderWidget->setSingleStep(1);
     }
   else
     {
-    d->BrushDiameterSlider->setMinimum(this->doubleParameter("BrushMinimumAbsoluteDiameter"));
-    d->BrushDiameterSlider->setMaximum(this->doubleParameter("BrushMaximumAbsoluteDiameter"));
-    d->BrushDiameterSlider->setValue(this->doubleParameter("BrushAbsoluteDiameter"));
-    d->BrushDiameterSlider->setSingleStep(this->doubleParameter("BrushMinimumAbsoluteDiameter"));
+    d->BrushDiameterSliderWidget->setRange(this->doubleParameter("BrushMinimumAbsoluteDiameter"), this->doubleParameter("BrushMaximumAbsoluteDiameter"));
+    d->BrushDiameterSliderWidget->setValue(this->doubleParameter("BrushAbsoluteDiameter"));
+    d->BrushDiameterSliderWidget->setSingleStep(this->doubleParameter("BrushMinimumAbsoluteDiameter"));
     }
-  d->BrushDiameterSlider->blockSignals(false);
-
-
-  d->BrushDiameterSpinBox->blockSignals(true);
-  d->BrushDiameterSpinBox->setMRMLScene(this->scene());
-  d->BrushDiameterSpinBox->setMinimum(d->BrushDiameterSlider->minimum());
-  d->BrushDiameterSpinBox->setMaximum(d->BrushDiameterSlider->maximum());
-  d->BrushDiameterSpinBox->setValue(d->BrushDiameterSlider->value());
   if (brushDiameterIsRelative)
     {
-    d->BrushDiameterSpinBox->setQuantity("");
-    d->BrushDiameterSpinBox->setSuffix("%");
-    d->BrushDiameterSpinBox->setDecimals(0);
+    d->BrushDiameterSliderWidget->setQuantity("");
+    d->BrushDiameterSliderWidget->setSuffix("%");
+    d->BrushDiameterSliderWidget->setDecimals(0);
     }
   else
     {
-    d->BrushDiameterSpinBox->setQuantity("length");
-    d->BrushDiameterSpinBox->setUnitAwareProperties(qMRMLSpinBox::Prefix | qMRMLSpinBox::Suffix);
-    int decimals = (int)(log10(d->BrushDiameterSlider->minimum()));
+    d->BrushDiameterSliderWidget->setUnitAwareProperties(qMRMLSliderWidget::Prefix | qMRMLSliderWidget::Suffix);
```

Comment:
Since `quantity` was set to empty, we need to restore it here. Or keep `quantity` property unchanged all the time and only change `unitAwareProperties` (and set `suffix` to `%` in relative mode).

Reply:
I've now restored the setting of quantity and suffix.